### PR TITLE
Use upstream KCFGShow class for CFG display

### PR DESCRIPTION
**Pull request by ehildenb**
Blocked on: https://github.com/runtimeverification/pyk/pull/258

This takes advantage of the upstreamed functionality in pyk for displaying KCFGs to:

- Shorten the code associated with displaying such information for the Foundry integration by reusing upstream `KCFGShow`.
- Factor out common argument parsers, including:
  - `display_args` for controlling the `--[no-]minimize option`
  - `spec_args` for loading specifications as KCFGs from file (and storing them).
  - `kcfg_show_args` for showing KCFGs.
- Offer KCFG display functionality for proofs run with `kevm prove ... --pyk ...` via new command `kevm show-kcfg ...`.
- Remove unused `minimize` argument to `foundry_step_node` and `foundry_section_edge`.
- Factor out `get_cfg_for_claim` which accepts the data from `spec_args` and extracts a single claim (as a KCFG) from it, by loading from disk.

This functionality was asked for by a user on Discord.

### CODE REVIEW

File: kevm-pyk/src/kevm_pyk/__main__.py [hunk 1]
```diff
@@ -299,6 +299,50 @@ def exec_prove(
...
 
 
+def exec_show_kcfg(
+    definition_dir: Path,
+    save_directory: Path,
+    spec_file: Path,
+    includes: List[str],
+    claim_label: Optional[str] = None,
+    spec_module: Optional[str] = None,
+    md_selector: Optional[str] = None,
```
Comment (Scott-Guest) [inline kevm-pyk/src/kevm_pyk/__main__.py:309]: Nitpick: The highlighted part here is exactly the signature for exec_view_kcfg, except that the order of 'save_directory' and 'spec_file' is flipped. Let's re-order them for consistency then.

File: kevm-pyk/src/kevm_pyk/__main__.py [hunk 2]
```diff
@@ -299,6 +299,50 @@ def exec_prove(
...
+            spec_module_name=spec_module,
+            include_dirs=[Path(i) for i in includes],
+            md_selector=md_selector,
+            claim_labels=([claim_label] if claim_label is not None else None),
+        )
+    )
+
+    kcfg = KCFGExplore.read_cfg(claim.label, save_directory)
+    if kcfg is None:
+        raise ValueError(f'Could not load CFG {claim} from {save_directory}')
```
Comment (Scott-Guest) [inline kevm-pyk/src/kevm_pyk/__main__.py:331]: These lines are a copy/paste from exec_view_kcfg. Is it worth pulling some of it out into a helper?

**Review by tothtamas28 — APPROVED** [review]

File: kevm-pyk/src/kevm_pyk/__main__.py [hunk 3]
```diff
@@ -769,6 +834,17 @@ def parse(s: str) -> List[T]:
...
+    show_kcfg_args = command_parser.add_parser(
+        'show-kcfg',
+        help='Display tree show of CFG',
+        parents=[shared_args, k_args, kcfg_show_args, display_args],
+    )
+    show_kcfg_args.add_argument(
+        'save_directory', type=dir_path, help='Path to where CFGs are stored (--save-directory option to prove).'
+    )
+    show_kcfg_args.add_argument('spec_file', type=file_path, help='Path to spec file.')
+    show_kcfg_args.add_argument('--claim', type=str, dest='claim_label', help='Claim identifier to load CFG for.')
```
Comment (Scott-Guest) [inline kevm-pyk/src/kevm_pyk/__main__.py:846]: These three arguments for 'view_kcfg_args' and 'show_kcfg_args' are the exact same. Should we pull them out into some shared 'kcfg_args'?

**Review by Scott-Guest — APPROVED** [review]
The changes look good to me! I commented on a few places where we can slightly reduce code duplication between show_kcfg and view_kcfg, but there are no real issues.

File: kevm-pyk/src/kevm_pyk/__main__.py (hunk 1 above)
Comment (ehildenb) [reply kevm-pyk/src/kevm_pyk/__main__.py]: Done.

File: kevm-pyk/src/kevm_pyk/__main__.py (hunk 2 above)
Comment (ehildenb) [reply kevm-pyk/src/kevm_pyk/__main__.py]: Done, Made a helper called `get_cfg_for_spec` in the `utils.py` module.

File: kevm-pyk/src/kevm_pyk/__main__.py (hunk 3 above)
Comment (ehildenb) [reply kevm-pyk/src/kevm_pyk/__main__.py]: I factored out `spec_args`, which I used for `prove_args`, `show_kcfg_args`, and `view_kcfg_args`. Now they all take exactly the same arguments.

Comment (ehildenb) [reply kevm-pyk/src/kevm_pyk/__main__.py]: Likely this `spec_args` should be factored out to upstream.

Comment (ehildenb) [reply kevm-pyk/src/kevm_pyk/__main__.py]: https://github.com/runtimeverification/pyk/issues/260

**Pull request merged**